src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -131,6 +131,15 @@ public static PhysicalAddress Parse(string address)
             {
                 hasDashes = true;
                 buffer = new byte[(address.Length + 1) / 3];
+
+                string[] parts = address.Split('-');
+                for (int i = 0; i < parts.Length; i++)
+                {
+                    if (parts[i].Length < 2)
+                    {
+                        throw new FormatException(SR.net_bad_mac_address);
+                    }
+                }
```

Comment:
As an aside it would be nice if the string included the bad Mac address. We are trying to include the bad value in exceptions from parsing, when practical, as often one only has the stack  trace in a log and no way to debug to see the value if it might be helpful.
No need to include in this PrR

Reply:
@danmosemsft I would like to create a PR to improve, according to your suggestion.
(It's sounds like a easy task for me 😅)
Currently the message is `An invalid physical address was specified.` .
Should it simply contain additional brackets with the given value or more information?

Reply:
Something similar to https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/Common/src/System/Net/Mail/MailAddressParser.cs#L232 add brackets on resource https://github.com/dotnet/corefx/blob/master/src/System.Net.NetworkInformation/src/Resources/Strings.resx#L77 and use `SR.Format` helper.

Reply:
Sure